### PR TITLE
add edge-driver code for hanssem smart switch

### PR DESCRIPTION
**Pull request by heoyslscomm**

### CODE REVIEW

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  local isParent = device.network_type ~= stDevice.NETWORK_TYPE_CHILD
+  local parent = isParent and device or device:get_parent_device()
+
+  local index
+
+  if isParent then 
+    index = device.fingerprinted_endpoint_id
+  else
+    index = tonumber(device.parent_assigned_child_key)
+  end
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:58]: all of this mapping should be handled by your `find_child` functions.

You may want to check the documentation here: https://developer.smartthings.com/docs/edge-device-drivers/zigbee/device.html#parent-child-devices

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua
[the remark this answers is not in the document]
Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:1]: add copyright notice

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua (hunk 1 above)
Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua]: I modified functions according to the guide.

Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua]: merge init.lua and utilities.lua into init.lua.

Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua]: I modified fingerprint.lua and init.lua by changing the model name

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  device:refresh()
+end
+
+--Handler Send Command 
+local function send_on_handle(driver, device)
+  if device.network_type ~= stDevice.NETWORK_TYPE_CHILD then
+    device:send(OnOff.server.commands.On(device):to_endpoint(device.fingerprinted_endpoint_id))
+  else
+    device:send(OnOff.server.commands.On(device):to_endpoint(tonumber(device.parent_assigned_child_key)))
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:76]: This extra `if` _should_ be unnecessary, I would expect. What happens when you delete the `:to_endpoint` and run the tests?

**Comment by heoyslscomm** [comment]
I delete unused variables and white space.

**Comment by heoyslscomm** [comment]
I removed unused variables and white space .

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua (hunk 2 above)
Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua]: Thank you for your guidance.
I removed unnecessary statements

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua
[the remark this answers is not in the document]
Comment (greens) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:1]: the current year is 2023

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  device:refresh()
+end
+
+local function send_on_handle(driver, device)
+  device:send_to_component(command.component, zclClusters.OnOff.server.commands.On(device))
+end
+
+local function send_off_handle(driver, device)
+  device:send_to_component(command.component, zclClusters.OnOff.server.commands.Off(device))
+end
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:75]: both of these duplicate default behavior and should be removed

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+local function attribute_handle(driver, device, value, zb_rx)
+  local attr = capabilities.switch.switch
+  device:emit_event_for_endpoint(zb_rx.address_header.src_endpoint.value, value.value and attr.on() or attr.off())
+end
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:80]: this also duplicates default behavior and should be removed

Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua]: I deleted functions that overlap with default behavior.

File: drivers/SmartThings/zigbee-switch/src/hanssem/init.lua
[the remark this answers is not in the document]
Comment (heoyslscomm) [reply drivers/SmartThings/zigbee-switch/src/hanssem/init.lua:1]: update copyright year

**Comment by greens** [comment]
Since your code coverage is now only at 65%, please add some tests to verify child device creation occurs as you expect. You will need to use the `device:expect_device_create` call in your tests.

**Review by greens — APPROVED** [review]

**Pull request merged**